comp/core/telemetry/component_mock.go

```diff
@@ -12,10 +12,21 @@ import (
 	sdk "go.opentelemetry.io/otel/sdk/metric"
 )
 
+// Metric interface defines the retrieval functions to extract information from a metric
+type Metric interface {
+	// GetTags returns the tags associated with the metric
+	GetTags() map[string]string
+	// GetValue returns the metric value
+	GetValue() float64
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** `Tags` and `Value` would be more idiomatic, but up to you if you want to change it